File: user/lrpstat/src/Debug.java
// Lrp Network Monitor: Simple network monitor written in Java
// Copyright (C) 2001 Martin Hejl
//
// This program is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 2, or (at your option)
// any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program; if not, write to the Free Software
// Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.

// $Revision: 1.6 $
// $Author: hejl $
// $Header: /home/cvs/lrpStat/src/Debug.java,v 1.6 2002/03/12 22:19:12 hejl Exp $

/**
 * Debug-Class that encapsulates debug-output
 * by setting DEBUG=false, all debugging output will be
 * turned off
 *
 * @author       Martin Hejl
 * @version      20020309 0.13beta
 */
public class Debug {
	protected static boolean DEBUG = true;

	public static void setVerbose(boolean b) {
		DEBUG=b;
	}
	public static void println(String s) {
		if (DEBUG) System.err.println(s);
	}

	public static void print(String s) {
		if (DEBUG) System.err.print(s);
	}

}

File: user/snort/src/debug.c
/* $Id$ */
/*
** Copyright (C) 1998-2002 Martin Roesch <roesch@sourcefire.com>
**
** This program is free software; you can redistribute it and/or modify
** it under the terms of the GNU General Public License as published by
** the Free Software Foundation; either version 2 of the License, or
** (at your option) any later version.
**
** This program is distributed in the hope that it will be useful,
** but WITHOUT ANY WARRANTY; without even the implied warranty of
** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
** GNU General Public License for more details.
**
** You should have received a copy of the GNU General Public License
** along with this program; if not, write to the Free Software
** Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
*/

#ifdef HAVE_CONFIG_H
#include "config.h"
#endif

#include <syslog.h>
#include <stdarg.h>
#include <stdlib.h>
#include <stdio.h>
#include "debug.h"

#include "snort.h"



#ifdef DEBUG


int debuglevel = DEBUG_ALL;
char *DebugMessageFile = NULL;
int DebugMessageLine = 0;

int DebugThis(int level)
{
    if (!(level & GetDebugLevel()))
    {
        return 0;
    }
    
    return 1;
}

int GetDebugLevel (void)
{
    static int debug_init = 0;
    static int debug_level = 0;

    if(debug_init) {
        return debug_level;
    }

    if (getenv(DEBUG_VARIABLE))
        debug_level = atoi(getenv(DEBUG_VARIABLE));
    else
        debug_level = 0;

    debug_init = 1;
    return debug_level;
}


void DebugMessageFunc(int level, char *fmt, ...)
{
    va_list ap;
    char buf[STD_BUF+1];


    if (!(level & GetDebugLevel()))
    {
        return;
    }

    /* filename and line number information */
    if (DebugMessageFile != NULL)
        printf("%s:%d: ", DebugMessageFile, DebugMessageLine);

    va_start(ap, fmt);
        
    if(pv.daemon_flag)
    {
        vsnprintf(buf, STD_BUF, fmt, ap);
        buf[STD_BUF] = '\0';
        syslog(LOG_DAEMON | LOG_DEBUG, "%s", buf);
    }
    else
    {
        vprintf(fmt, ap);
    }

    va_end(ap);
}
#else
void DebugMessageFunc(int level, char *fmt, ...)
{
}
#endif /* DEBUG */
